### PR TITLE
added mpm_event and php-fpm support

### PR DESCRIPTION
**Pull request by PiyushRaj927**
Fixes issue #54 
This PR adds support for Apache mpm_event and PHP_FPM 

## testing
PHP-FPM is being used as shown below 

brave_03-15-23(17-44-471)

During manually setting up the wiki as per [docs](https://canasta.wiki/setup/#create-new-wiki), I came across an unexpected behavior, when copying the generated `LocalSettings.php` in the `/config` directory and restarting the docker containers does not update the `/var/www/mediawiki/w/LocalSettings.php` file. The updated file is present at `/mediawiki/config/LocalSettings.php`.
I had to manually update the file for setting up the wiki.
The wiki is shown below

### CODE REVIEW

**Comment by PiyushRaj927** [comment]
@jeffw16 please review my pull request

**Review by jeffw16 — CHANGES_REQUESTED** [review]

File: _sources/scripts/run-apache.sh [hunk 1]
```diff
@@ -223,6 +223,10 @@ jobrunner &
 transcoder &
 sitemapgen &
 
+# Running php-fpm
+service php7.4-fpm start
```
[the remark this answers is not in the document]
Comment (PiyushRaj927) [reply _sources/scripts/run-apache.sh:227]: Thanks for the suggestions @jeffw16, but to use [systemd in a non-privileged container](https://developers.redhat.com/blog/2016/09/13/running-systemd-in-a-non-privileged-container#the_quest) we will need to mount the host cgroup file system. I suggest using the [init](https://docs.docker.com/compose/compose-file/#init) provided by docker and launching php-fpm and Apache from a shell script as given here https://docs.docker.com/config/containers/multi-service_container/
process tree is shown below based on the above proposed method.
```bash
$ pstree
docker-init-+-apachectl-+-apache2-+-apache2
            |           |         |-3*[apache2---26*[{apache2}]]
            |           |         `-3*[rotatelogs]
            |           |-run-apache.sh---runuser---mwjobrunner.sh---sleep
            |           `-run-apache.sh---runuser---mwtranscoder.sh---sleep
            `-php-fpm7.4---2*[php-fpm7.4]```

[the remark this answers is not in the document]
Comment (jeffw16) [reply _sources/scripts/run-apache.sh:227]: Your last proposal that isn't about using `service` or `systemctl` is great; not sure why we didn't just go for that in the first place. If you could please implement that into your changes to run PHP-FPM, that'd be great, thanks.

**Comment by PiyushRaj927** [comment]
I have implemented the discussed changes; please review them as well as this related PR [Canasta-DockerCompose#41](https://github.com/CanastaWiki/Canasta-DockerCompose/pull/41)

**Comment by PiyushRaj927** [comment]
@jeffw16 please review the edits

**Comment by jeffw16** [comment]
Hi, sorry, I had a busy weekend. The edits look good. What testing have you done for this? @PiyushRaj927

**Comment by jeffw16** [comment]
> During manually setting up the wiki as per [docs](https://canasta.wiki/setup/#create-new-wiki), I came across an unexpected behavior, when copying the generated LocalSettings.php in the /config directory and restarting the docker containers does not update the /var/www/mediawiki/w/LocalSettings.php file. The updated file is present at /mediawiki/config/LocalSettings.php.
I had to manually update the file for setting up the wiki.

This is concerning. You had to update the LocalSettings.php file manually inside the container?

**Comment by PiyushRaj927** [comment]
Hi @jeffw16 hope your weekend went well.

> What testing have you done for this after my initial PR review?

I have tested the changes by creating a sample wiki and checking the output of `<?php phpinfo();?>`  like my initial PR.
Is there anything else I need to test to confirm that the modifications don't cause any issues. 

> This is concerning. You had to update the LocalSettings.php file manually inside the container?

Yes, I had to run `cp /mediawiki/config/LocalSettings.php /var/www/mediawiki/w/LocalSettings.php` in the container for setting up the wiki after placing the `LocalSettings.php` in `/config` and restarting the stack.

**Comment by jeffw16** [comment]
> Yes, I had to run cp /mediawiki/config/LocalSettings.php /var/www/mediawiki/w/LocalSettings.php in the container for setting up the wiki after placing the LocalSettings.php in /config and restarting the stack.

Thanks for clarifying. This is a blocker. I'm guessing that the PHP-FPM cache is acting up. In this case, we will need to take a look at the startup process to see why this is happening and fix it.

**Comment by PiyushRaj927** [comment]
@jeffw16, I looked into the startup process and finally found that by default, environment variables are not accessible to the php-fpm workers (see [clear_env](https://www.php.net/manual/en/install.fpm.configuration.php)). I have disabled this behavior.
I have also tested the changes and did not notice any issues.

Furthermore, I have attached the screenshot of a test wiki I made 

brave_03-21-23(23-11-183)
The `Powered by Canasta` footer is now being rendered, which was not the case earlier.

**Comment by jeffw16** [comment]
I just tried it out. This is blazing fast! Awesome job @PiyushRaj927! We'll be merging it in shortly.

**Review by jeffw16 — APPROVED** [review]
LGTM

**Comment by yaronkoren** [comment]
@PiyushRaj927 - great work! And thank you.

**Pull request merged**